hw/ip/prim/doc/prim_flash.md

```diff
@@ -1,157 +1,159 @@
-When the erase suspend completes, the flash wrapper circuitry also asserts `done` for the ongoing erase transaction to ensure all hardware gracefully completes.
-
-The following is an example diagram
-{{< wavejson >}}
-{signal: [
-  {name: 'clk_i',                wave: 'p................'},
-  {name: 'pg_erase_i',           wave: '01............0..'},
-  {name: 'ack_o',                wave: '1.0..............'},
-  {name: 'erase_suspend_i',      wave: '0.....1.......0..'},
-  {name: 'done_o',               wave: '0............10..'},
-  {name: 'erase_suspend_done_o', wave: '0............10..'},
-]}
-{{< /wavejson >}}
+---
+title: "Primitive Component: Flash Wrapper"
+---
+
+# Overview
+`prim_flash` is a wrapper interface for technology specific flash modules.
+
+As the exact details of each technology can be different, this document mainly describes the interface requirements and their functions.
+The wrapper however does assume that all page sizes are the same (they cannot be different between data and info partitions, or different types of info partitions).
+
+## Parameters
+
+Name           | type   | Description
+---------------|--------|----------------------------------------------------------
+NumBanks       | int    | Number of flash banks.  Flash banks are assumed to be identical, asymmetric flash banks are not supported
+InfosPerBank   | int    | Maximum number of info pages in the info partition.  Since info partitions can have multiple types, this is max among all types.
+InfoTypes      | int    | The number of info partition types, this number can be 1~N.
+InfoTypesWidth | int    | The number of bits needed to represent the info types.
+PagesPerBank   | int    | The number of pages per bank for data partition.
+WordsPerPage   | int    | The number of words per page per bank for both information and data partition.
+DataWidth      | int    | The full data width of a flash word (inclusive of metadata)
+MetaDataWidth  | int    | The metadata width of a flash word
+TestModeWidth  | int    | The number of test modes for a bank of flash
+
+
+## Signal Interfaces
+
+### Overall Interface Signals
+Name                    | In/Out | Description
+------------------------|--------|---------------------------------
+clk_i                   | input  | Clock input
+rst_n_i                 | input  | Reset input
+flash_req_i             | input  | Inputs from flash protocol and physical controllers
+flash_rsp_o             | output | Outputs to flash protocol and physical controllers
+prog_type_avail_o       | output | Available program types in this flash wrapper: Currently there are only two types, program normal and program repair
+init_busy_o             | output | The flash wrapper is undergoing initialization
+tck_i                   | input  | jtag tck
+tdi_i                   | input  | jtag tdi
+tms_i                   | input  | jtag tms
+tdo_o                   | output | jtag tdo

```

Comment:
is there supposed to be a trst_ni here also?
@arnonsha had mentioned it i think.

Reply:
We've checked and there is no need for trst_ni. trst is generated by bist internally from rst_n.

Reply:
@tjaychen 
My bad, I looked at bist_top which has a trst port, but then again this is not actually the top level :-)
In the real top level the trst_n signal is set to 1 after powerup (inactive), and **doesn't come from an ext pad**